draft-ietf-capport-api.md

```diff
@@ -130,7 +130,11 @@ Upon receiving this information the client will provide this information to the
 
 # Security Considerations {#security}
 
-TBD: Provide complete security requirements and analysis.
+One of the goals of this protocol is to improve the security of the communication between client hosts and Captive Portal systems. Client traffic is protected from passive listeners on the local network by requiring TLS-encrypted connections between the client and the Captive Portal API server, as described in {{api-details}}. All communication between the clients and the API server MUST be encrypted.
+
+In addition to ecrypting communications between clients and Captive Portal systems, thie protocol requires a basic level of authentication from the API server, as described in {{server-auth}}. Specifically, the API server MUST present a valid certificate on which the client can perform revocation checks. This allows the client to ensure that the API server has authority for a hostname that can be presented to a user. This hostname MUST be presented to the user when associating with the Captive Portal, in order to benefit from the authentication check.
+
+It is important to note that while the server authentication checks can validate a specific hostname, it is certainly possible for the API server to present a valid certificate for a hostname that uses non-standard characters or is otherwise designed to trick the user into believing that its hostname is some other, more trustworthy, name. This is a danger of any scenario in which a hostname is not typed in by a user. In order to mitigate this attack, client systems MAY choose to whitelist certain hostnames that they expect to see as captive portals (such as being pre-provisioned to trust "airline.example.com"), and potentially block or warn the user about unknown hostnames.
```

Comment:
This bit is challenging.  I would hesitate to recommend any specific response here.  Specificity might encourage simplistic responses that could be easily circumvented, or it might discourage the seeking of alternative and superior approaches.  It is enough to highlight the fact that the name offered by the network might be constructed in a way as to confuse recipients about identity.

For me, I would prefer not to present these names directly (in contradiction to the previous paragraph) on the basis that they are not relevant to user activities, except in support of other features.  For instance, rendering the info page in a web view might be done without offering the name of the server.  But that's something for designers to grapple with; all you need to do is help them understand the properties of the system.

Reply:
That's a fair point; for now, I've just removed the suggested mitigation and left the consideration that the name itself may be bogus.